### PR TITLE
Add trim test using devices supporting and not supporting discards

### PR DESCRIPTION
**Pull request by katlapinka**
- Add test using devices supporting and not supporting discards
- Move test_trim_start_discard from /trim to /io/trim group

Needs:
https://github.com/Open-CAS/test-framework/pull/183

Signed-off-by: Katarzyna Lapinska <katarzyna.lapinska@intel.com>

### CODE REVIEW

**Comment by katlapinka** [comment]
Need redesign

File: test/functional/tests/io/trim/test_trim.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        TestRun.disks["hdd"].create_partitions([Size(1, Unit.GibiByte)])
+        ssd1_dev = TestRun.disks["ssd1"].partitions[0]
+        ssd2_dev = TestRun.disks["ssd2"].partitions[0]
+        hdd_dev = TestRun.disks["hdd"].partitions[0]
+
+    with TestRun.step(f"Start cache and add core."):
+        cache_dev = ssd1_dev if discard_support["cache"] else hdd_dev
+        core_dev = ssd2_dev if discard_support["core"] else hdd_dev
+
+        cache = casadm.start_cache(cache_dev, cache_mode)
```
Comment (Ostrokrzew) [inline test/functional/tests/io/trim/test_trim.py:124]: Add force parameter, please.

File: test/functional/tests/io/trim/test_trim.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    with TestRun.step("Stopping cache"):
+        cache.stop()
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode.with_traits(CacheModeTrait.InsertWrite))
+@pytest.mark.parametrizex("filesystem", Filesystem)
+@pytest.mark.parametrizex("cleaning_policy", CleaningPolicy)
+@pytest.mark.parametrizex("discard_support", [{"cache": False, "core": True},
+                                              {"cache": True, "core": False},
+                                              {"cache": True, "core": True}])
```
Comment (Ostrokrzew) [inline test/functional/tests/io/trim/test_trim.py:92]: At this moment this cannot be dict (see: https://github.com/Open-CAS/test-framework/issues/210). For now You should change it to some immutable structure, e.g. tuple. And don't forget to update usage of `discard_support` according to changes.

**Review by Ostrokrzew — CHANGES_REQUESTED** [review]

File: test/functional/tests/io/trim/test_trim.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        TestRun.LOGGER.info(f"CAS {property_name} value is correct.")
+        return
+    TestRun.LOGGER.error(f"CAS property {property_name} value equals {value} and differs "
+                         f"from expected value: {expected_value}.")
+
+
+def stop_monitoring_and_check_discards(blktraces, discard_support):
+    time.sleep(10)
+    os_utils.sync()
+    os_utils.drop_caches()
```
Comment (Ostrokrzew) [inline test/functional/tests/io/trim/test_trim.py:196]: Please, add 1-2 seconds of sleep after `drop_caches()` - some DUTs, especially with RAIDs, need it.

**Review by Ostrokrzew — CHANGES_REQUESTED** [review]

**Review by Ostrokrzew — APPROVED** [review]

**Review by Deixx — DISMISSED** [review]

**Review by Deixx — APPROVED** [review]

**Review by robertbaldyga — APPROVED** [review]

**Pull request merged**